### DISCUSSION TITLE
Question when installing via pip

### DISCUSSION THREAD

**Discussion by flog123** [martin-ueding/geo-activity-playground#190] [Q&A] [answered]
The focus of my request is not really regarding geo-activity-playground, its more a pip issue. But as i was faced with it although I  strictly followed your instruction for the [stable version on linux](https://martin-ueding.github.io/geo-activity-playground/getting-started/installing-stable-on-linux/), I guess other users will run into this too.

When I enter

`pip install --user geo-activity-playground`

I receive the following message.

```
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.12/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
```

Therefore I decided to install geo-activity-plugin via pipx, as it is stated to be the easiest use...

`pipx install --user geo-activity-playground` 

But pipx does not recognise the --user argument.

```
pipx install --user geo-activity-playground
usage: pipx [-h] [--quiet] [--verbose] [--version]
            {install,uninject,inject,upgrade,upgrade-all,uninstall,uninstall-all,reinstall,reinstall-all,list,run,runpip,ensurepath,environment,completions}
            ...
pipx: error: unrecognized arguments: --user
```

I could have left the --user argument, but I don´t think thats in the sense of the author.

Does anyone have an idea?

**Comment by martin-ueding** [comment, marked as answer]
Python's package management unfortunately is quite convoluted. What is going on here is that Debian wants to prevent you from installing arbitrary Python packages into the system wide directory which is managed by the APT package manager. And that makes sense.

I find it peculiar that `pip install --user` doesn't work here. But okay.

`pipx` always installs into the local user's directory, so it doesn't have a `--user` flag. You should be able to just do the following:

```
pipx install geo-activity-playground
```

**Comment by beautiful-orca** [comment]
When i installed it on linux, i used conda/python environments to install dependencies. The docker version takes out the hassle, tho.
